### PR TITLE
Feat: tool links tests

### PR DESCRIPTION
**Pull request by sorodrigo**
This PR adds unit tests to tool-links reducer.

- [ ] finish testing select nodes by search edge cases
- [ ] test select nodes

### CODE REVIEW

File: frontend/scripts/react-components/tool-layers/tool-layers.reducer.js [hunk 1]
```diff
@@ -90,19 +83,21 @@ const toolLayersReducer = {
...
       draft.data.mapContextualLayers = {};
-      action.mapContextualLayers.forEach(layer => {
+      mapContextualLayers.forEach(layer => {
         draft.data.mapContextualLayers[layer.id] = layer;
       });
     });
   },
   [TOGGLE_MAP_DIMENSION](state, action) {
     return immer(state, draft => {
+      console.log('do,', action.payload.selectedMapDimensions);
```
Comment (sorodrigo) [inline frontend/scripts/react-components/tool-layers/tool-layers.reducer.js]: Boop

File: frontend/scripts/react-components/tool-links/tool-links.actions.js [hunk 2]
```diff
@@ -19,6 +19,7 @@ export const TOOL_LINKS__SET_SELECTED_BIOME_FILTER = 'TOOL_LINKS__SET_SELECTED_B
 export const TOOL_LINKS_SET_NO_LINKS_FOUND = 'TOOL_LINKS_SET_NO_LINKS_FOUND';
 export const TOOL_LINKS_RESET_SANKEY = 'TOOL_LINKS_RESET_SANKEY';
 export const TOOL_LINKS__SET_MISSING_LOCKED_NODES = 'TOOL_LINKS__SET_MISSING_LOCKED_NODES';
+export const SET_SELECTED_NODES_BY_SEARCH = 'SET_SELECTED_NODES_BY_SEARCH';
```
Comment (sorodrigo) [inline frontend/scripts/react-components/tool-links/tool-links.actions.js]: we should follow the namespace convention

File: frontend/scripts/tests/tool-links/reducer.test.js [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+        }
+      }
+    };
+    const action = selectSearchNode(results);
+    const newState = reducer(state, action);
+    expect(newState).toMatchSnapshot();
+  });
+
+  // TODO:
+  // - REGRESSION TEST: selectedColumnsIds = [empty, 2, empty, empty] and select a node that has column group === 4 and column.isDefault === true
```
Comment (sorodrigo) [inline frontend/scripts/tests/tool-links/reducer.test.js]: We should add this test before merging

**Comment by Bluesmile82** [comment]
Ready!
